actions/image_partition_action.go

```diff
@@ -584,6 +586,61 @@ func (i ImagePartitionAction) PostMachineCleanup(context *debos.DebosContext) er
 	return nil
 }
 
+func ParseOffset(offset string) (int, string, error) {
+	/* Extract value of offset (int) and type from string*/
+	var v, t []rune
+	for _, l := range offset {
+		switch {
+		case l >= '0' && l <= '9':
+			v = append(v, l)
+		case l >= 'A' && l <= 'Z', l >= 'a' && l <= 'z', l == '%':
+			t = append(t, l)
+		}
+	}
+
+	val, err := strconv.Atoi(string(v))
+	typ := string(t)
+
+	if err != nil {
+		return 0, "", fmt.Errorf("Can not parse %s to integer", string(v))
+	}
+
+	return val, typ, nil
+}
+
+func CalculateOffset(start, end string) (string, error) {
+	/* Do addition if end value uses a '+' prefix */
+	if strings.HasPrefix(end, "+") {
```

Comment:
Nit: I would move the "HasPrefix" check in the caller.

Reply:
I agree.